src/GRA.Controllers/AvatarController.cs

```diff
@@ -59,12 +59,70 @@ public async Task<IActionResult> Index()
                 .GroupBy(_ => _.GroupId)
                 .Select(_ => _.ToList())
                 .ToList();
+            var usersresult = _avatarService.GetUserUnlockBundles().Result;
+            var userbundles = usersresult.Item1;
+            var hasbeenviewed = usersresult.Item2;
+            var bundles = new AvatarBundleJsonModel
+            {
+                Bundles = _mapper
+                .Map<List<AvatarBundleJsonModel.AvatarBundle>>(userbundles)
+            };
             var viewModel = new AvatarViewModel
             {
                 LayerGroupings = layerGroupings,
+                Bundles = userbundles,
+                ViewedBundles = hasbeenviewed,
                 DefaultLayer = userWardrobe.First(_ => _.DefaultLayer).Id,
                 ImagePath = _pathResolver.ResolveContentPath($"site{GetCurrentSiteId()}/avatars/"),
-                AvatarPiecesJson = Newtonsoft.Json.JsonConvert.SerializeObject(model)
+                AvatarPiecesJson = Newtonsoft.Json.JsonConvert.SerializeObject(model),
+                AvatarBundlesJson = Newtonsoft.Json.JsonConvert.SerializeObject(bundles)
+            };
+
+            var userAvatar = await _avatarService.GetUserAvatarAsync();
+            if (userAvatar.Count == 0)
+            {
+                viewModel.NewAvatar = true;
+            }
+            return View(viewModel);
+        }
+
+        public async Task<IActionResult> BundleIndex()
+        {
+            var userWardrobe = await _avatarService.GetUserWardrobeAsync();
+            if (userWardrobe.Count == 0)
+            {
+                ShowAlertDanger("Avatars have not been configured.");
+                _logger.LogError("User {id} tried to customize their avatar but avatars have not been configured!",
+                    GetActiveUserId());
+                return RedirectToAction(nameof(HomeController.Index), HomeController.Name);
+            }
+
+            var model = new AvatarJsonModel
+            {
+                Layers = _mapper
+                .Map<ICollection<AvatarJsonModel.AvatarLayer>>(userWardrobe)
+            };
+            var layerGroupings = userWardrobe
+                .GroupBy(_ => _.GroupId)
+                .Select(_ => _.ToList())
+                .ToList();
+            var usersresult = _avatarService.GetUserUnlockBundles().Result;
```

Comment:
This also needs to be an async call if the method is `async` and the tuple can be assigned as above (`var (userBundles, hasBeenViewed) = await _avatarService.GetUserUnlockBundles();`).